products/libllbuild/BuildSystem-C-API.cpp

```diff
@@ -381,6 +381,17 @@ class CAPIBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
       free((char *)rules[i].key);
     }
   }
+
+  virtual void error(StringRef filename, const Token& at, const Twine& message) override {
+    if (cAPIDelegate.handle_diagnostic) {
+      cAPIDelegate.handle_diagnostic(cAPIDelegate.context,
+                                     llb_buildsystem_diagnostic_kind_error,
+                                     filename.str().c_str(),
+                                     -1,
+                                     -1,
+                                     message.str().c_str());
+    }
+  }
```

Comment:
Should this fall back to the default implementation if the client doesn't define a handler?

Reply:
That sounds like a good idea, will update.